### PR TITLE
Fix: DO-2171 cleanup of local storage

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

It was found that at some point local storage was running out of space. Seems like we were not cleaning up session related values.

## Implementation Description

Added so that we clean session related objects from Local storage.

## How Has This Been Tested?

Locally by using `Variable`s with persist flag and checking that storage was cleaned after log out/log in

Also updated a test for cleaning cache. Before it falsely succeeded due to waitFor not being properly awaited.

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [x] I have added relevant tests (unit, integration or regression).
- [x] I have added comments to all the bits that are hard to follow.
- [ ] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

### Before
not clean

### After
clear storage

### CODE REVIEW

File: packages/dara-core/js/shared/utils/clean-session-cache.tsx [hunk 1]
```diff
@@ -6,13 +6,21 @@
  */
 function cleanSessionCache(sessionToken: string): void {
     const sessionKeys = Object.keys(sessionStorage);
+    const localKeys = Object.keys(localStorage);
```
Comment (krzysztof-causalens) [inline packages/dara-core/js/shared/utils/clean-session-cache.tsx:9]: Minor but this could be simplified if we just loop through the two storages, e.g.

```tsx
for (const storage of [sessionStorage, localStorage]) {
 const keys = Object.keys(storage);
 ...
}
```

File: packages/dara-core/tests/js/template-root.spec.tsx [hunk 2]
```diff
@@ -31,21 +31,21 @@ describe('TemplateRoot', () => {
         expect(getByText('Frame, Menu')).toBeInstanceOf(HTMLSpanElement);
     });
 
-    it('should clean up cache on startup', () => {
+    it('should clean up cache on startup', async () => {
```
Comment (krzysztof-causalens) [inline packages/dara-core/tests/js/template-root.spec.tsx:34]: This test doesn't cover `sessionStorage` cleanup, could we have it cover both storages (or add another one to cover it)

**Review by krzysztof-causalens — CHANGES_REQUESTED** [review]

File: packages/dara-core/tests/js/template-root.spec.tsx (hunk 2 above)
Comment (patricia-causalens) [reply packages/dara-core/tests/js/template-root.spec.tsx:34]: oops had forgotten to add sessionStorage, has been added now

**Review by krzysztof-causalens — APPROVED** [review]
